src/main/java/org/opensearch/flowframework/common/DefaultUseCases.java

```diff
@@ -22,94 +32,119 @@ public enum DefaultUseCases {
+        "substitutionTemplates/deploy-remote-model-extra-params-template.json",
+        List.of(CREATE_CONNECTOR_CREDENTIAL_KEY)
     ),
     /** defaults file and substitution ready template for Bedrock Titan embedding model */
     BEDROCK_TITAN_EMBEDDING_MODEL_DEPLOY(
         "bedrock_titan_embedding_model_deploy",
         "defaults/bedrock-titan-embedding-defaults.json",
-        "substitutionTemplates/deploy-remote-bedrock-model-template.json"
+        "substitutionTemplates/deploy-remote-bedrock-model-template.json",
+        List.of(CREATE_CONNECTOR_CREDENTIAL_ACCESS_KEY, CREATE_CONNECTOR_CREDENTIAL_SECRET_KEY, CREATE_CONNECTOR_CREDENTIAL_SESSION_TOKEN)
     ),
     /** defaults file and substitution ready template for Bedrock Titan multimodal embedding model */
     BEDROCK_TITAN_MULTIMODAL_MODEL_DEPLOY(
         "bedrock_titan_multimodal_model_deploy",
         "defaults/bedrock-titan-multimodal-defaults.json",
-        "substitutionTemplates/deploy-remote-bedrock-model-template.json"
+        "substitutionTemplates/deploy-remote-bedrock-model-template.json",
+        List.of(CREATE_CONNECTOR_CREDENTIAL_ACCESS_KEY, CREATE_CONNECTOR_CREDENTIAL_SECRET_KEY, CREATE_CONNECTOR_CREDENTIAL_SESSION_TOKEN)
     ),
     /** defaults file and substitution ready template for Cohere chat model */
     COHERE_CHAT_MODEL_DEPLOY(
         "cohere_chat_model_deploy",
         "defaults/cohere-chat-defaults.json",
-        "substitutionTemplates/deploy-remote-model-chat-template.json"
+        "substitutionTemplates/deploy-remote-model-chat-template.json",
+        List.of(CREATE_CONNECTOR_CREDENTIAL_KEY)
     ),
     /** defaults file and substitution ready template for OpenAI chat model */
     OPENAI_CHAT_MODEL_DEPLOY(
         "openai_chat_model_deploy",
         "defaults/openai-chat-defaults.json",
-        "substitutionTemplates/deploy-remote-model-chat-template.json"
+        "substitutionTemplates/deploy-remote-model-chat-template.json",
+        List.of(CREATE_CONNECTOR_CREDENTIAL_KEY)
     ),
     /** defaults file and substitution ready template for local neural sparse model and ingest pipeline*/
     LOCAL_NEURAL_SPARSE_SEARCH_BI_ENCODER(
         "local_neural_sparse_search_bi_encoder",
         "defaults/local-sparse-search-biencoder-defaults.json",
-        "substitutionTemplates/neural-sparse-local-biencoder-template.json"
+        "substitutionTemplates/neural-sparse-local-biencoder-template.json",
+        Collections.emptyList()
     ),
     /** defaults file and substitution ready template for semantic search, no model creation*/
-    SEMANTIC_SEARCH("semantic_search", "defaults/semantic-search-defaults.json", "substitutionTemplates/semantic-search-template.json"),
+    SEMANTIC_SEARCH(
+        "semantic_search",
+        "defaults/semantic-search-defaults.json",
+        "substitutionTemplates/semantic-search-template.json",
+        List.of(CREATE_INGEST_PIPELINE_MODEL_ID)
```

Comment:
Not a part of this PR but can you create an issue to improve `semantic-search-template` to include creating model as well?

Reply:
@joshpalis is this being tracked anywhere? I see the comment is resolved

Reply:
we have a `semantic-search-with-model` template which includes model registration